homeassistant/components/hassio/sensor.py

```diff
@@ -0,0 +1,122 @@
+    ATTR_VERSION,
+    ATTR_VERSION_LATEST,
+    SUPERVISOR,
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_MONITORED_VARIABLES, default=ALL_UPDATABLE_ITEMS): vol.All(
+            cv.ensure_list, [vol.In(ALL_UPDATABLE_ITEMS)]
+        )
+    }
+)
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: Config,
+    async_add_entities_callback: Callable[[List[Entity], bool], None],
+    discovery_info=None,
+):
+    """Set up the Hass.io sensor platform."""
+    async_add_entities_callback([HassioUpdatesSensor(config[CONF_MONITORED_VARIABLES])])
+
+
+class HassioUpdatesSensor(Entity):
+    """Hass.io updates sensor."""
+
+    def __init__(self, types_to_track: List[str]) -> None:
+        """Initialize sensor."""
+        self._types_to_track = types_to_track
+        self._available_updates = []
+        self._unsub_listener = None
+
+    async def _update(self) -> None:
+        """Update sensor data."""
+        self._available_updates = []
+        if ADDONS in self._types_to_track:
+            self._available_updates = [
+                {**addon, ATTR_TYPE: ADDON}
+                for addon in self.hass.data[DATA_INFO][SUPERVISOR].get(ATTR_ADDONS, [])
+                if addon[ATTR_INSTALLED] and addon[ATTR_UPDATE_AVAILABLE]
+            ]
+        for type in self._types_to_track:
+            if type == ADDONS:
+                continue
+            data = {**self.hass.data[DATA_INFO][type], ATTR_TYPE: type}
+            if data.get(ATTR_UPDATE_AVAILABLE, False):
+                self._available_updates.append(data)
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        self._unsub_listener = async_dispatcher_connect(
+            self.hass, f"{DATA_INFO}_update", self._update
+        )
+        await self._update()
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity will be removed from hass."""
+        if self._unsub_listener:
+            self._unsub_listener()
+            self._unsub_listener = None
+
+    @property
+    def name(self) -> str:
+        """Return sensor name."""
+        return "Hass.io: Updates"
+
+    @property
+    def state(self) -> int:
+        """Return state of entity."""
+        return len(self._available_updates)
+
+    @property
+    def device_state_attributes(self) -> Dict[str, Any]:
+        """Return device specific state attributes."""
+        data = []
+        for update in self._available_updates:
+            item = {ATTR_TYPE: update[ATTR_TYPE]}
+            if update[ATTR_TYPE] == ADDON:
+                item[ATTR_NAME] = update[ATTR_NAME]
+                item[ATTR_SLUG] = update[ATTR_SLUG]
+            item[ATTR_AVAILABLE_VERSION] = update[ATTR_VERSION_LATEST]
+            item[ATTR_INSTALLED_VERSION] = update[ATTR_VERSION]
+            data.append(item)
+        return {ATTR_UPDATES: data}
```

Comment:
We should not use attributes for these things, instead make separate sensors.

And for a lot of things, it doesn't matter. For example, the Supervisor, CLI, Observer, audio (actually all, except core & add-ons); are auto-updated in the background.

Reply:
OK, should we have separate sensors for each add-on as well?

Reply:
I think so? I would say, we create separate sensors/switch/entities for each add-on and/or other information we have, and decide on which ones we disable by default (so we keep the most common/useful ones).

Reply:
does OS get updated in the background too? I'm guessing not. I could add a device and sensors for OS as well.

Reply:
No the OS won't get updates automatically

Reply:
OK I can add that in a follow up PR 👍🏾

Reply:
Since this didn't make the beta, I will go ahead and add that to this PR

Reply:
Since this PR is still open, I just implemented it as part of this